src/plugins/intel_gna/transformations/convert_unaligned_concat_to_cnn.cpp

```diff
@@ -0,0 +1,535 @@
+    const auto values = std::vector<uint32_t>{ static_cast<uint32_t>(e1), static_cast<uint32_t>(e2) };
+    return std::make_shared< ngraph::opset7::Constant>(ov::element::u32, ngraph::Shape{ 2 }, values);
+}
+
+template <typename T>
+std::shared_ptr < ngraph::opset7::Constant > CreateConstantU64V(const std::vector<T>& values) {
+    return ngraph::opset7::Constant::create(ov::element::u64, ngraph::Shape{ values.size() }, values);
+}
+
+static std::string rtValue(size_t val) {
+    return std::to_string(val);
+}
+
+template <typename N, typename K>
+std::shared_ptr < ov::Node > AppendStridedSliceOn2DNode(
+    std::shared_ptr<ov::Node> node, N dim2Begin, K dim2End, bool addForbidFC = true) {
+    std::vector<int64_t> mask{};
+    IE_ASSERT(node->get_output_size() == 1);
+    auto shape = node->get_output_shape(0);
+    IE_ASSERT(shape.size() == 2);
+    constexpr auto expectedDim1 = 1;
+    IE_ASSERT(shape[0] == expectedDim1);
+    if (dim2Begin == 0 && dim2End == shape[1]) {
+        return node;        // don't append crop as not needed
+    }
+    auto begin = GetConstant2U(0, dim2Begin);
+    auto end = GetConstant2U(expectedDim1, dim2End);
+    auto outputNode = std::make_shared < ngraph::opset7::StridedSlice >(node, begin, end, mask, mask);
+
+    if (addForbidFC && dim2Begin != 0) {
+        // supress substitution of this StridedSlice with affine layer
+        // this operation is handled by pointer arithmetics and the unnedeed part (outside outputNode output) will be overwriten
+        outputNode->get_rt_info()[GNAConcatLayer::GetForbidFC()] = rtValue(1);
+    }
+    return outputNode;
+}
+
+static std::shared_ptr< ngraph::opset7::MatMul> AppendAffineCopy(std::shared_ptr<ov::Node> node,
+    size_t numberOfInputs, size_t numberOfOutputs,
+    size_t firstInputIndex, size_t firstOutputIndex) {
+    IE_ASSERT(node->get_output_size() == 1);
+    IE_ASSERT(node->get_output_shape(0).size() == 2);
+    IE_ASSERT(node->get_output_shape(0)[0] == 1);
+    IE_ASSERT(node->get_output_shape(0)[1] == numberOfInputs);
+    const auto elementType = node->get_output_element_type(0);
+    auto values = GetCopyAffineFilterValues(numberOfInputs, numberOfOutputs,
+        firstInputIndex, firstOutputIndex);
+    auto shape = ngraph::Shape{ numberOfInputs, numberOfOutputs };
+    auto filter = std::make_shared< ngraph::opset7::Constant>(elementType, shape, values);
+    auto appended = std::make_shared< ngraph::opset7::MatMul>(node, filter);
+    // This RT param is not preserved at the graph compiler
+    appended->get_rt_info()["UnalignedConcatMatMul"] = rtValue(1);
+    return appended;
+}
+
+static void PutRtConcat(std::shared_ptr<ov::Node> node, size_t offset, size_t size) {
+    node->get_rt_info()["concat_output_offset"] = rtValue(offset);
+    node->get_rt_info()["concat_effective_size"] = rtValue(size);
+}
+
+static void DisableCF(std::shared_ptr<ov::Node> node) {
+    node->get_rt_info()[ov::pass::DisableConstantFolding::get_type_info_static()];
```

Comment:
As we discussed previously right way is using `ov::disable_constant_folding` function. As far as I understood this attitude should not be copyable. You should mark appropriate nodes directly. cc @GlebKazantaev Could you please comment?

Reply:
Any updates ?
Currently, all the other problems were addressed.

Reply:
Yes, please use `ov::disable_constant_folding`

Reply:
@GlebKazantaev The problem is that when I use `ov::disable_constant_folding`
two tests are failing
which does not happen for 
`node->get_rt_info()[ov::pass::DisableConstantFolding::get_type_info_static()];`

Reply:
@GlebKazantaev Any update or timeline regarding this?

Reply:
@kbruniec Which updates are you waiting for from @GlebKazantaev? I think we need to understand the root cause of failing tests. Why do you need it copyable?

Reply:
@GlebKazantaev "Asked for some time" as he needs to analyze why the disable_constant_folding is not working.
We were analyzing/debugging some tests and despite disable_constant_folding was applied
such nodes were folded anyway (by `ngraph::pass::ConvertOpSet1ToLegacy`). 

I don't fully understand why the `copyable` property matters, but the test is working fine when `copyable` is "enabled".
I don't insist to make it copyable, it is just a clue from my experiments

Reply:
@kbruniec Thank you for the clarification. So exactly these marked nodes are fused?

Reply:
Yes, we tried to mark Constant nodes as well as Reshape nodes and no success - they were folded anyway.